src/accel/cuda/GPUSolver.cu

```diff
@@ -1454,18 +1452,54 @@ void GPUSolver::transportSweep() {
   /* Initialize flux in each FSR to zero */
   flattenFSRFluxes(0.0);
 
-  /* Sweep the first halfspace of azimuthal angle space */
-  tid_offset = 0;
-  tid_max = (_tot_num_tracks / 2);
+  /* Sweep the first space of azimuthal angle and periodic track halfspaces */
+  tid_offset = _num_tracks_by_halfspace[0];
```

Comment:
Again, I missed this during the initial review, but think that this function bloat could be conslidated into a nice for loop.